tfjs-backend-webgpu/src/kernels/BatchNormal.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {KernelConfig, FusedBatchNorm, FusedBatchNormInputs,FusedBatchNormAttrs, Tensor} from '@tensorflow/tfjs-core';
+import {WebGPUBackend} from '../backend_webgpu';
+import {BatchNormProgram} from './batchnorm_webgpu';
+
+export const batchNromConfig : KernelConfig = {
+    kernelName: FusedBatchNorm,
+    backendName: 'webgpu',
+    kernelFunc: ({inputs,attrs,backend}) => {
+        const {x,scale,offset,mean,variance} = inputs as FusedBatchNormInputs;
+        const {varianceEpsilon} = attrs as unknown as FusedBatchNormAttrs;
+        const webGPUBackend = backend as WebGPUBackend;
+        let batchNormInputs = [x as Tensor, mean as Tensor, variance as Tensor];
+        let offsetShape = null;
+        if(offset != null) {
+            offsetShape = offset.shape;
+            batchNormInputs.push(offset as Tensor);
+        }
+        let scaleShape = null;
+        if(scale != null) {
+            scaleShape = scale.shape;
+            batchNormInputs.push(scale as Tensor);
+        }
+        const program = new BatchNormProgram(
+            x.shape,mean.shape,variance.shape,offsetShape,scaleShape,
```

Comment:
Could you install our code formatter clang? Otherwise whitespace diffs will appear in later commits :)

Details for installation here: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md